ci_scripts/generate_news.py

```diff
@@ -1,64 +1,71 @@
-# You may obtain a copy of the License at
-#
-# http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-# --------------------------------------------------------------------------
-"""Handles usage of towncrier for automated changelog generation
-
-a) abstracts configuration and execution for this project
-b) prepares for usage in CI environment to stop from generating
-   entries for every minor patch version
-
-However, when deploying to production, we always want a changelog
 """
+Handles usage of towncrier for automated changelog generation
+Handles usage of pyautoversion for versioning
+"""
+
 import os
 import subprocess
+import logging
+import enum
+import argparse
+from auto_version import auto_version_tool
 
-from distutils.version import LooseVersion
-
-PROJECT_ROOT = os.path.dirname(os.path.dirname(__file__))
-news_dir = os.path.join(PROJECT_ROOT, 'docs', 'news')
-metafile = os.path.join(news_dir, 'last_built.meta')
+logger = logging.getLogger(__name__)
 
+PROJECT_ROOT = os.path.join(os.path.dirname(os.path.dirname(__file__)), "..")
+PROJECT_CONFIG = os.path.join(os.path.dirname(__file__), "pyproject.toml")
+NEWS_DIR = os.path.join(PROJECT_ROOT, 'news')
 
-def main():
-    """Writes out newsfile if significant version bump"""
-    last_known = '0'
-    if os.path.isfile(metafile):
-        with open(metafile) as fh:
-            last_known = fh.read()
 
-    import mbed_cloud
-    current = mbed_cloud.__version__
+class CommitType(enum.Enum):
+    DEVELOPMENT = 1
+    BETA = 2
+    RELEASE = 3
 
-    # how significant a change in version scheme should trigger a new changelog entry
-    # (api major, api minor, sdk major, sdk minor, sdk patch)
-    sigfigs = 4
-    current_version = LooseVersion(current).version
-    last_known_version = LooseVersion(last_known).version
 
-    should_towncrier = current_version[:sigfigs] != last_known_version[:sigfigs]
+def version(commit_type):
+    """
+    Versions the project
+    :param commit_type: states what is the type of the commit
+    :type: CommitType
+    """
+    BUMP_TYPES = {CommitType.DEVELOPMENT: "build",
+                  CommitType.BETA: "prerelease"}
 
-    print('%s -- %s :: current vs previous changelog build' % (current, last_known))
-    if should_towncrier:
-        print('%s >> %s :: running changelog build' % (current, last_known))
+    old, new, updates = auto_version_tool.main(
+        release=commit_type == CommitType.RELEASE,
+        enable_file_triggers=True if commit_type in [CommitType.BETA,
+                                                     CommitType.RELEASE] else None,
+        bump=BUMP_TYPES.get(commit_type, None),
```

Comment:
`{}.get()` returns `None` by default, so it's not necessary to pass `None` as `default` arg.